packages/sdk/src/types.ts

```diff
@@ -109,39 +102,33 @@ export interface NormalizedOrder<OriginalOrder> {
   originalOrder: OriginalOrder;
 }
 
-export type OpenseaNormalizedOrder = NormalizedOrder<OpenseaOrder>;
+export type OpenseaOrder = NormalizedOrder<OpenseaOriginalOrder>;
 
-export type TraderNormalizedOrder = NormalizedOrder<TraderOrder>;
+export type TraderOrder = NormalizedOrder<TraderOriginalOrder>;
 
-export type LooksRareNormalizedOrder = NormalizedOrder<LooksRareOrder>;
+export type LooksRareOrder = NormalizedOrder<LooksRareOriginalOrder>;
 
-interface ResponseData<N, D> {
+interface ResponseData<N extends MarketplaceName, D> {
   marketplaceName: N;
-  data: D;
+  data?: D;
```

Comment:
A `ResponseData` with an optional `data` is kinda 😑 

How about
```typescript
interface ResponseData<N extends MarketplaceName, D> {
  marketplaceName: N;
  data: D;
}
interface ResponseError<N extends MarketplaceName> {
  marketplaceName: N;
  error: {
    message: string;
  };
}
type Response<N extends MarketplaceName, D> = Merge<
  ResponseData<N, D>,
  ResponseError<N>
>;
```

Reply:
Can't find any ways to do
```typescript
if (resp.error) {
  // handle error
  return
}
// do something with resp.data // resp.data must exist here
```

So more Golang style of handling errors.

Reply:
Was originally using
```ts
// either-or version
type Response<N extends MarketplaceName, D> =
  | ResponseData<N, D>,
  | ResponseError<N>;
```
so either data or error is guaranteed

But for some reason it causes problems in the union
```ts
type OrderResponse =
  | OpenseaOrderResponse
  | TraderOrderResponse
  | LooksRareOrderResponse;
```
where the type of `marketplaceName` field is somehow inferred as `MarketplaceName.LooksRare` instead of `MarketplaceName`
Not sure why

Good news is.. If and when we manage to get the either-or version working, we can always publish a new version with that swapped in without causing a breaking change